c++/src/capnp/compat/json.c++

```diff
@@ -379,13 +382,208 @@ void JsonCodec::encodeField(StructSchema::Field field, DynamicValue::Reader inpu
   encode(input, field.getType(), output);
 }
 
+namespace {
+int64_t parseInt64(kj::StringPtr s) {
+  char *endPtr;
+  errno = 0;
+  int64_t value = std::strtoll(s.begin(), &endPtr, 10);
+  KJ_REQUIRE(endPtr == s.end(), "String does not contain valid number", s);
+  KJ_REQUIRE(errno != ERANGE, "Value out-of-range", s);
```

Comment:
I don't see point to complicate the code. Do we really care about strtoll implementation that returns errno==ERANGE  and it does not mean range error? I consider it invalid strtoll implementation. My interpretation is that strltoll is required to not set errno if conversion was successful.